torch/csrc/jit/passes/quantization.cpp

```diff
@@ -23,7 +72,30 @@ void InsertObserverNodes(std::shared_ptr<Graph>& graph) {
 }
 
 void InsertFakeQuantNodes(std::shared_ptr<Graph>& graph) {
-  throw std::runtime_error("Pass not implemented yet!");
+  for (const auto& n : graph->nodes()) {
+    // Skip nodes that we've just added
+    if ((n->kind().toQualString() == std::string("aten::quant")) ||
+      (n->kind().toQualString() == std::string("aten::dequant"))) {
+      continue;
+    }
+    if (checkIfNodeValuesQuantizable(n) != 0) {
+      for (torch::jit::Value* v : n->outputs()) {
+        if (v->isCompleteTensor()) {
+          addQuantDeQuantNodes(v);
```

Comment:
We could use `WithInsertPoint`- we won't need to pass the node into `addQuantDeQuantNodes` then. Also we can return the created value from this function and move `replaceInputWith`/`replaceAllUsesWith` out of it - this should make the function simpler.

Reply:
WithInsertPoint throws error while mutating the input as for input node v->input belong to different block. replaceInputWith()/replaceAllUsesWith() need to happen before we set v as input to quant. Moreover i feel the logic might scatter outside this api. For now we will retain this as is

Reply:
I have slightly modified logic. The special case node inputs will kick in only for external inputs